### PR TITLE
webpack

### LINKED ISSUE #204: Modularize source files
As an F2 developer, I'd like to modularize the source files to make compilation simpler, explicitly call out dependencies, and set F2 up to be broken up in the future.

The typical approaches include:
1. `CommonJS --> Browserify`: Simple and straightforward.
2. `ES6 --> ES6 Transpiler --> CommonJS or AMD --> Browserify`: Future proof, but complicated to implement.
3. `AMD + loader --> Grunt concat`: Simple, but exposes AMD modules and requires bundled AMD loader.
4. `Globals --> Grunt concat`: (current) This requires special linting rules to accommodate implied globals and unused variables.

My personal choice would be to start with "1", which will give us an easy win without much effort. From there, it will be more straightforward to switch to something like "2" in the future.

### PR DESCRIPTION
**Pull request by brianbaker**
migrate codebase to use webpack for build. fixes #204

### CODE REVIEW

File: .jshintrc [hunk 1]
```diff
@@ -1,4 +1,5 @@
 {
+	"esversion": 9,
```
Comment (pc2459) [inline .jshintrc:2]: o brave new world

File: tests/spec/utils-spec.js [hunk 2]
```diff
@@ -0,0 +1,41 @@
+describe('F2 utility methods', function() {
+
+	beforeEachReloadF2();
+
+	it('F2.version should return a version number (SemVer as String)', function() {
+		//https://github.com/sindresorhus/semver-regex
+		var testVersion = /\bv?(?:0|[1-9][0-9]*)\.(?:0|[1-9][0-9]*)\.(?:0|[1-9][0-9]*)(?:-[\da-z\-]+(?:\.[\da-z\-]+)*)?(?:\+[\da-z\-]+(?:\.[\da-z\-]+)*)?\b/ig.test(F2.version());
```
Comment (pc2459) [inline tests/spec/utils-spec.js:7]: `semver` package used by npm would be saner but something something ain't broke don't fix https://www.npmjs.com/package/semver

File: webpack.config.js [hunk 3]
```diff
@@ -0,0 +1,64 @@
...
+const path = require('path');
+const pkg = require('./package.json');
+const webpack = require('webpack');
+
+module.exports = {
+	devtool: 'source-map',
+	entry: './src/index.js',
+	mode: 'production',
+	optimization: {
+		minimize: false
```
Comment (pc2459) [inline webpack.config.js:11]: If `grunt`'s no longer uglifying, then what is for `.min`? Or are you just not generating one?

File: webpack.config.js [hunk 4]
```diff
@@ -0,0 +1,64 @@
...
+			name: 'F2',
+			type: 'umd'
+		}
+	},
+	plugins: [
+		new webpack.BannerPlugin({
+			banner: () => {
+				return `
+F2 v${pkg.version} ${new Date().toLocaleString('en', { dateStyle: 'short' })}
+Copyright (c) 2014 Markit On Demand, Inc. http://www.openf2.org
```
Comment (pc2459) [inline webpack.config.js:27]: Should this be 2021 and use the correct modern company name?

**Review by pc2459 — APPROVED** [review]

**Comment by pc2459** [comment]
Hotel Califorrrrniaaaaa ✔️

File: webpack.config.js (hunk 3 above)
Comment (brianbaker) [reply webpack.config.js:11]: I'll eventually let `minimize` be `true`.  I left it as false for now because I wanted to see the output :-).

**Pull request merged**